File: frontend_Angular/eventos_peru/src/app/modelos/usuario.ts
// Modelo base del usuario
export interface Usuario {
  idUsuario: number | null; // opcional al crear (autoincrementa en DB)
  nombre: string;
  email: string;
  password?: string; // <-- IMPORTANTE: nombre que espera el backend
  rol: 'ADMIN' | 'CLIENTE' | 'PROVEEDOR';
  fechaRegistro?: string | Date; // admitimos string 'YYYY-MM-DD' o Date
  celular?: string;
}

File: backend _Spring Boot/backend/src/main/java/com/eventosperu/backend/model/Usuario.java
package com.eventosperu.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "usuarios")
public class Usuario {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idUsuario;

    @Column(nullable = false, length = 100)
    private String nombre;

    @Column(nullable = false, unique = true, length = 100)
    private String email;
    

    @Column(name = "telefono", length = 20)

    private String celular;

    @Column(nullable = false, length = 255)
    private String password;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Rol rol;

    private LocalDateTime fechaRegistro = LocalDateTime.now();

    public enum Rol {
        ADMIN,
        CLIENTE,
        PROVEEDOR
    }
}
